WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -359,23 +359,22 @@ final class ProductFormViewController<ViewModel: ProductFormViewModelProtocol>:
                 break
             case .variations(let row):
                 ServiceLocator.analytics.track(.productDetailViewVariationsTapped)
-                guard let originalProduct = viewModel.originalProductModel as? EditableProductModel, row.isActionable else {
+                guard row.isActionable else {
                     return
                 }
-                let variationsViewModel = ProductVariationsViewModel(formType: viewModel.formType)
-                let variationsViewController = ProductVariationsViewController(initialViewController: self,
-                                                                               viewModel: variationsViewModel,
-                                                                               product: originalProduct.product)
-                variationsViewController.onProductUpdate = { [viewModel] updatedProduct in
-                    viewModel.updateProductVariations(from: updatedProduct)
+                showVariations()
+            case .noPriceWarning(let viewModel):
+                ServiceLocator.analytics.track(.productDetailViewVariationsTapped)
+                guard viewModel.isActionable else {
+                    return
```

Comment:
I think this guard should go before the analytics, or we will be login taps on un-tappable rows, like on the product variation view.


https://user-images.githubusercontent.com/562080/135634944-e1d65664-8fba-472f-8288-b2bff10b4f2e.mov

Reply:
Thanks for reviewing @Ecarrion ! ❤️ 
You are right. I left it this way, because we were previously logging the event before the guard. I will fix it.